libs/parallelism/execution/include/hpx/execution/detail/post_policy_dispatch.hpp

```diff
@@ -69,19 +72,21 @@ namespace hpx { namespace parallel { namespace execution { namespace detail {
                 using is_void = typename std::is_void<result_type>::type;
                 hpx::detail::call_sync(
                     is_void{}, std::forward<F>(f), std::forward<Ts>(ts)...);
-                return;
+                return threads::invalid_thread_id;
             }
 
             threads::thread_init_data data(
                 threads::make_thread_function_nullary(hpx::util::deferred_call(
                     std::forward<F>(f), std::forward<Ts>(ts)...)),
                 desc, priority, hint, stacksize,
                 threads::thread_schedule_state::pending);
-            threads::register_work(data);
+
+            threads::thread_id_ref_type thrd = threads::register_work(data);
+            return hint.runs_as_child ? thrd : threads::invalid_thread_id;
```

Comment:
I have reverted this particular change, post will now return `void` as before.

Reply:
What consequences did reverting this have? What did you have to change to make up for it?

Reply:
No big consequences. I have changed the implementation of these function to support returning the thread_id: https://github.com/STEllAR-GROUP/hpx/pull/5467/files#diff-6abd893418c71fd43f8d01620d9c601f3f12f4d1e987e9a78144669e3ff5f54aR154-R183

Reply:
The main thing to be aware of with these changes is that threads will be created eagerly (though not necessarily the stack) for . I don't know if that's a good, neutral, or bad thing.

If I understand correctly, this would apply to `async`, `then`, and `dataflow` *without* an explicit executor, correct?

Reply:
Yes, you're right on both accounts. There is also the issue that the scoped execution is currently enabled through the `bool` in the thread-hint, but we don't have any way to pass that to the non-executor APIs, but OTOH, the scoped execution should be applied to the non-executor APIs only ... More design work is needed, apparently.